File: frontend/src/constants/ShipStatus.js
export const ShipStatus = Object.freeze({
    IDLE:           'IDLE',
    LOADING:        'LOADING',
    LOADED:         'LOADED',
    IN_TRANSIT:     'IN_TRANSIT',
    AWAITING_PILOT: 'AWAITING_PILOT',
    CUSTOMS_HOLD:   'CUSTOMS_HOLD',
    MINIGAME_WAITING: 'MINIGAME_WAITING',
    MINIGAME_ACTIVE: 'MINIGAME_ACTIVE',
    SEIZED:         'SEIZED',
});

export function isDocked(status) {
    return status === ShipStatus.IDLE || status === ShipStatus.LOADING || status === ShipStatus.LOADED;
}

export function isAtSea(status) {
    return status === ShipStatus.IN_TRANSIT || status === ShipStatus.AWAITING_PILOT || status === ShipStatus.MINIGAME_WAITING || status === ShipStatus.MINIGAME_ACTIVE;
}

export function isCustomsHold(status) {
    return status === ShipStatus.CUSTOMS_HOLD;
}

export function isSeized(status) {
    return status === ShipStatus.SEIZED;
}

File: backend/service-game-engine/src/main/java/at/fhv/engine/dto/ShipStatus.java
package at.fhv.engine.dto;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum ShipStatus {
    IDLE,
    LOADING,
    LOADED,
    UNLOADING,
    IN_TRANSIT,
    AWAITING_PILOT,
    CUSTOMS_HOLD,
    MINIGAME_WAITING,
    MINIGAME_ACTIVE,
    SEIZED;

    @JsonCreator
    public static ShipStatus fromJson(String value) {
        if (value == null) return null;
        try {
            return ShipStatus.valueOf(value);
        } catch (IllegalArgumentException ex) {
            return null;
        }
    }
}